modin/engines/ray/pandas_on_ray/frame/partition.py

```diff
@@ -395,26 +458,38 @@ def deploy_ray_func(call_queue, partition):  # pragma: no cover
     def deserialize(obj):
         if isinstance(obj, ObjectIDType):
             return ray.get(obj)
-        return obj
+        elif isinstance(obj, (tuple, list)) and any(
+            isinstance(o, ObjectIDType) for o in obj
+        ):
+            return ray.get(list(obj))
+        elif isinstance(obj, dict) and any(
+            isinstance(val, ObjectIDType) for val in obj.values()
+        ):
+            return dict(zip(obj.keys(), ray.get(list(obj.values()))))
+        else:
+            return obj
 
-    if len(call_queue) > 1:
-        for func, kwargs in call_queue[:-1]:
+    if len(funcs) > 1:
```

Comment:
why do we need this `if` and why do we special-case the last entry in the list?..

Reply:
I left this code as it was but yes, we can rid of special-casing the last entry in the list.

Reply:
Fixed